pkg/utils/k8sutil.go

```diff
@@ -747,3 +759,33 @@ func addInitContainerAnnotation(dpm *v1beta1.Deployment) error {
 	}
 	return nil
 }
+
+func getKV(label string) (string, string) {
+	character := ""
+	equalPos := strings.Index(label, "=")
+	colonPos := strings.Index(label, ":")
+	switch {
```

Comment:
I _think_ what you're trying to do here is to split the string on the first `:` or `=`.

To simplify this greatly, `strings.IndexAny` will look for the first occurrence of several characters in one step, and note that the `equalPos`/`colonPos` already tells us where to split the string, we don't need need to search for the delimiter again:
```go
func getKV(input string) (string, string) {
   var key, value string
   if pos := strings.IndexAny(input, "=:"); pos != -1 {
      key = input[:pos]
      value = input[pos+1:]
   } else {
      // no separator found
     key = input
     value = ""
   }

   if value == "" {
      value = os.Getenv(key)
   }

   return key, value
}
```